### ISSUE TITLE
Analytics KPI dashboard request for [VANotify]

### ISSUE THREAD

**Issue by miabecker** [department-of-veterans-affairs/va.gov-team#34550]
# Request an Edge Case KPI Dashboard

## What this form is for

Use this template to request a KPI dashboard on Domo for your product. This template is for products that fall outside of our standardized dashboards.

If you have another VA.gov product, please reach out to the Platform Analytics & Insights team on #vfs-platform-support.

### Expected delivery timelines

Please note that these dashboards will require additional data and design discovery, and will take additional time to deliver. 

Understood - we would ideally like this stood up by the end of January 2022. Open to feedback and insight on what sort of timelines this dashboard will take.

## Instructions for the VFS team

- [X] **Please fill out [this template](https://docs.google.com/spreadsheets/d/1TSynjEUrOrm_HfLxb2NAPDXxnn7wyy1bSEb2RruC67Q/edit#gid=0) and link it below to help define your product/feature's information.** _(Please request the Excel sheet from the Platform Analytics & Insights team if you do not have Google access.)_ The Analytics team will ***not*** be able to pull this ticket into the queue until this is provided and linked.

https://docs.google.com/spreadsheets/d/1tYevYp34WT22b73pFp-fU9v_gNJEc5p1xWFNe5ZmNwQ/edit?usp=sharing

The form and content areas of the spreadsheet do not seem to be relevant to our ask but open to feedback. Our data on notification statuses will be coming from Big Query and exported into DOMO. See query below.

```
DECLARE STARTDATE DATE;
DECLARE ENDDATE DATE;
DECLARE SERVICENAME STRING;

SET STARTDATE =  '2022-01-31';
SET ENDDATE = '2022-02-05';
SET SERVICENAME = 'dmc';


select FORMAT_DATE('%A, %b %d', results.date) as Date, results.Requests, results.NotFoundInMPI, results.MarkedAsDeceasedInMPI, results.NoVAProfileIDInMPI, results.NoEmailVaProfile, results.NotFoundInVAProfile,
results.Requests - (results.NotFoundInMPI + results.MarkedAsDeceasedInMPI + results.NoVAProfileIDInMPI + results.NoEmailVaProfile + results.NotFoundInVAProfile) as SuccessfulEmailAddressRetrieval,
Concat(
    cast(
        round(
        (results.Requests - 
            (results.NotFoundInMPI + results.MarkedAsDeceasedInMPI + results.NoVAProfileIDInMPI + results.NoEmailVaProfile + results.NotFoundInVAProfile)
        ) / results.Requests * 100, 2) 
        as string), 
    '%') as SuccessfulEmailAddressRetrievalRate,
results.Delivered as SuccessfulEmailDelivery, 
Concat(
    cast(
        round(
            results.Delivered / (results.Requests - (results.NotFoundInMPI + results.MarkedAsDeceasedInMPI + results.NoVAProfileIDInMPI + results.NoEmailVaProfile + results.NotFoundInVAProfile)) * 100, 2) 
        as string), 
    '%') as SuccessEmailDeliveryRate
FROM (SELECT 
date, 
SUM(count) as Requests,
SUM(CASE WHEN status_reason = 'Mpi Profile not found for this identifier' THEN count ELSE 0 END) as NotFoundInMPI,
SUM(CASE WHEN status_reason = 'Beneficiary has deceased status in MPI' THEN count ELSE 0 END) as MarkedAsDeceasedInMPI,
SUM(CASE WHEN status_reason = 'Requested identifier not found in MPI correlation database' THEN count ELSE 0 END) as NoVAProfileIDInMPI,
SUM(CASE WHEN status_reason = 'No contact info found from VA Profile' THEN count ELSE 0 END) as NoEmailVaProfile,
SUM(CASE WHEN status_reason = 'Received Forbidden HTTP error (403) while making a request to obtain info from VA Profile' THEN count ELSE 0 END) as NotFoundInVAProfile,
SUM(CASE WHEN status = 'delivered' THEN count ELSE 0 END) as Delivered
FROM `vsp-analytics-and-insights.platform_vanotify.prod-statistics` where service_name=SERVICENAME and (date >= STARTDATE and date <= ENDDATE) 
group by date
order by date) as results;
```

  - [X] If a part of your product involves any sort of linear process/funnel for which you want to be able to measure conversions, please include it in the "Forms" tab of the template.

N/A 

 - [X] All other pages associated with your product for which you'd want to track visits and engagement should be added to the "Content" tab.
 

- [X] **Add links** to product outlines, feature information, or any other resources you think would help us better understand your product and OKRs to the "VFS-provided information" section below.

- [X] **Add your team label** to this ticket to follow progress.

- [X] If you do not have **Domo access**, please [request access here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=joanneesteban&labels=analytics-insights%2C+analytics-request%2C+access-request&template=analytics-request-google-analytics-domo-access.md&title=Request+access+to+Google+Analytics+and%2For+Domo).

## VFS-provided information

Add links to product outlines, feature information, or any other resources you think would help us better understand your product and OKRs below. Feel free to also type background information into this section directly. The more context you're able to provide, the fewer questions we'll need to ask you before getting started!

#### Product resources
- VANotify is a centralized notifications platform serving VA.gov as well as business lines across the VA. You can learn more by visiting our github page linked below.
- https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/va-notify
- We want to provide insight into our notification statistics to our business line and VA.gov customers by providing data in a DOMO dashboard.  

#### Other product info
_(Add here as needed.)_
The open and click-thru numbers that we provide, through GA, for our business lines are listed below.  We would like to have these added to the dashboard and have the dashboard calculate open/click-thru rates

- [open numbers](https://analytics.google.com/analytics/web/#/report/content-event-events/a50123418w177519031p176188361/_u.date00=20211201&_u.date01=20220104&_r.drilldown=analytics.eventCategory:email,analytics.eventAction:open&explorer-table.plotKeys=%5B%5D&explorer-table.secSegmentId=analytics.eventLabel&explorer-table.rowCount=10&explorer-segmentExplorer.segmentId=analytics.eventAction/)
- [click-thru numbers for claim-appeal-status](https://analytics.google.com/analytics/web/#/report/content-landing-pages/a50123418w177519031p176188361/_u.date00=20211201&_u.date01=20220104&explorer-table.plotKeys=%5B%5D&explorer-table.advFilter=%5B%5B0,%22analytics.source%22,%22PT%22,%22vanotify%22,0%5D%5D&explorer-table.secSegmentId=analytics.campaign&explorer-table.rowCount=100&_r.drilldown=analytics.landingPagePath:www.va.gov~2Fclaim-or-appeal-status~2F&explorer-segmentExplorer.segmentId=analytics.source/)
- [click-thru numbers for manage-va-debt](https://analytics.google.com/analytics/web/#/report/content-landing-pages/a50123418w177519031p176188361/_u.date00=20211201&_u.date01=20220104&_r.drilldown=analytics.landingPagePath:www.va.gov~2Fmanage-va-debt~2F&explorer-segmentExplorer.segmentId=analytics.source&explorer-table.plotKeys=%5B%5D&explorer-table.filter=vanotify&explorer-table.secSegmentId=analytics.campaign&explorer-table.advFilter=%5B%5B0,%22analytics.source%22,%22RE%22,%22vanotify%22,0%5D%5D/)

**Additional Resources from the Platform Analytics team**

- Please use the [KPI Framework](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/analytics/Analytics%20Playbook/va-gov-platform-analytics-kpi-framework.pdf) for more information on the metrics used in the Standard KPI dashboards.

- For additional info, read through the [How to Request a KPI Dashboard Documentation (https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Analytics-customer-support-guidelines.1586823275.html#Analyticscustomersupportguidelines-4.RequestaKPIDashboardinDomo).

*Please leave the following blank*

------

## VSP Analytics & Insights Acceptance Criteria

1. Discovery & Design

- [ ] Determine appropriate product questions
- [ ] Determine appropriate metrics & charts
- [ ] Design dashboard layout

2. Build (See [internal process document](https://vfs.atlassian.net/wiki/spaces/AT/pages/1464598569/New+KPI+Dashboard+Request+Process) & [how to](https://vfs.atlassian.net/wiki/spaces/AT/pages/1538294010/WIP%2B-%2BHow%2Bto%2Bcreate%2Bstandardized%2Bdashboards%2Bin%2BDomo))

- [ ] Add the `VSP Analytics & Insights` Group as the dashboard owner
- [ ] Structure funnel data for ETL
- [ ] Pipe available metrics to Domo
- [ ] Document any blocked/unavailable metrics in the Analytics team space
- [ ] Update the dashboard header, including Google Analytics links
- [ ] If any, add [implementation issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=joanneesteban&labels=analytics-insights%2C+analytics-request%2C+collaboration-cycle%2C+collab-cycle-review%2C+gtm&template=analytics-implementation-and-qa-request-template.md&title=Analytics+Implementation+or+QA+Support+for+[Team+Name+-+Feature+Name]) for additional necessary implementation

3. Q&A

- [ ] Please see & complete the [checklist for full KPI dashboarding QA checklist](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsp/teams/insights-analytics/analytics-tools/domo-dashboard-qa-checklist.md)

4. Share

- [ ] Add VFS group & default group to the dashboard

*For new VFS customers only:*

- [ ] Schedule onboarding meeting

- [ ] Host the meeting to explain how the dashboard works

## Definition of Done

Acceptance Criteria has been completed

- [ ] KPI Dashboard full QA checklist completed
- [ ] Dashboard has been shared back with the VFS team

**Comment by miabecker** [comment]
@chloedotbrown just wanted to check in on this ticket. Thank you!

**Comment by chloedotbrown** [comment]
Hi @miabecker – thanks for reaching out! Analytics will discuss this in our sprint planning meeting tomorrow, now that our full team is back in the office. 

It does look like this will take custom KPI-setting and design work, and may potentially require some new data pipelines. There are also a couple edge case requests in our queue ahead of this one, so getting this stood up by the end of the month may not be something we're able to do. We should be able to update with a better estimate after discussing tomorrow.

**Comment by miabecker** [comment]
Adding our data fields 
[VA Notify Data.pdf](https://github.com/department-of-veterans-affairs/va.gov-team/files/7824123/VA.Notify.Data.pdf)

**Comment by bevnobev** [comment]
@michelle-dooley, Maria added the fields that we are planning to upload to BigQuery.  We are awaiting @bsmartin-ep create a bucket for us in BigQuery so we can upload to S3

**Comment by miabecker** [comment]
@michelle-dooley wanted to check in and see if you needed anything else to identify where this data will be leaving and creating the edgecase dashboard?

**Comment by michelle-dooley** [comment]
@miabecker I think we have what we need for now, we just need to dig into it a bit and see where this fits in with our other work.  Will let you know as soon as we have more to share.

**Comment by obrienjulie** [comment]
@miabecker - Hi Maria, I have created a new table in Big Query based on the data fields that you supplied earlier. The new table is vsp-analytics-and-insights:platform_vanotify.email_results.

**Comment by miabecker** [comment]
@obrienjulie we are good I think on pushing the data into big query - tomorrow might be our first day in production. We are waiting on an edge case dashboard and have a ticket in for that.

**Comment by bevnobev** [comment]
Adding status definitions - https://docs.notifications.service.gov.uk/rest-api.html#email-status-descriptions

**Comment by chloedotbrown** [comment]
Thanks so much for adding this, @bevnobev – and do let us know if you're able to get open rates out of AWS as well! It'll give us a more accurate count than the GA data, if it's available.

**Comment by bevnobev** [comment]
@jakehova @nathanbwright Are we able to get open rates from AWS?  see above for context

**Comment by jakehova** [comment]
We will look into this.

**Comment by mjones-oddball** [comment]
Status Definitions:

- Sent: message was sent to the provider but it has NOT been received by the recipient yet
- Delivered: message was received by the recipient
- Sending: message is enqueued for send but hasnt been sent by the provider yet
- Technical Failure:
-- max price is exceeded (highly unlikely to happen)
-- invalid message (highly unlikely to happen since we are using templates to generate the message)
-- general invalid (if the recipient phone number is invalid)
- Temporary Failure: cant get in touch with provider or carrier and will try again soon OR waited to long to get a Sent status and will try again soon
- Permanent Failure: phone number has been blocked or marked as spam

**Comment by mjones-oddball** [comment]
@bevnobev posted status definitions in comments and added the query from Jacob into the ticket description. If I missed anything please let me know! Per our discussion, we won't be getting open rates from AWS.

**Comment by chloedotbrown** [comment]
Hi @bevnobev - wanted to follow up on our conversation in office hours about this ticket. Our team is working on the dashboard design, and we've set up pipelines to bring the Google Analytics and notification data into Domo (updated to reflect query above). However, after reviewing the data, we're not going to be able to create an accurate open-rate estimate using a combination of the two. If open-rates do become available through the AWS data, then we'd love to use those. But for now, we'll have to show those metrics as total open events. 

We're looking forward to connecting with @mjones-oddball and the new VANotify team soon to sync up on data sources and make sure the MVP matches their needs. I believe @michelle-dooley will be reaching out soon!

**Comment by bevnobev** [comment]
Thanks Michelle.  I updated the spreadsheet with one more ask, I link it's line 14.  It's the click-thru rate for va.gov/profile.  cc: @mjones-oddball

**Comment by mjones-oddball** [comment]
@mjones-oddball testing notification

**Comment by mjones-oddball** [comment]
@mjones-oddball test

**Comment by mjones-oddball** [comment]
@mjones-oddball test

**Comment by mjones-oddball** [comment]
@mjones-oddball

**Comment by chloedotbrown** [comment]
Hi @bevnobev - here's some language for cautioning folks against trying to recreate open-rates using GA data. I've attached a screenshot as well. Hope this helps!

> We recommend against trying to create "open rates" using totals from Google Analytics. Because this data captures users who clicked from a campaign sent at any time prior to the date range selected, these totals will almost always be much higher than the true number of users who clicked from campaigns sent during that period. This results in artificially high open rates, sometimes over 100%. 
> 
> For example, in a comparison of GA and notification data in the same period (1/25-2/5), email opens were higher than notifications delivered in 5 out of 11 cases. 

<img width="1306" alt="Open Rates" src="https://user-images.githubusercontent.com/45518242/154594304-bc9ba082-8dbc-456f-ba2c-503b0817abad.png">

**Comment by mjones-oddball** [comment]
@michelle-dooley Just wanted to check in and see if we're on track for end of March still? Anything you need from us?

**Comment by michelle-dooley** [comment]
@mjones-oddball - yep we are still on track for end of March!!

**Comment by michelle-dooley** [comment]
@bevnobev - I see your access was granted.  Are you able to see it now?

You are correct on bounce rate.  If you ever have questions on metric definitions you can always click on the card title and it will take you to a view that allows you to see more information about how it was calculated (looks like a typo in this one, will get that fixed) and drill down into details.

![image](https://user-images.githubusercontent.com/93597659/161296043-ffc836b4-a528-4a16-a1f9-18839827f059.png)

**Comment by bevnobev** [comment]
@michelle-dooley Thanks!  I have access now.

**Comment by bevnobev** [comment]
@michelle-dooley Seems like the stats are only until 3/26, but we write to big query nightly.

**Comment by michelle-dooley** [comment]
Hi @bevnobev that is correct.  Domo dashboards default groupings and comparisons are week over week so we generally only load once per week so you are comparing apples to apples.  For example if we included the week of 3/27 you would see a big drop in this chart b/c the week is not over (may not now since it is Friday but earlier in the week you would have).  You can still see daily trends by changing the graph by to day at the top right but it will still only report through 3/26.  This week's data will be loaded Sunday.  If you would like us to change this to daily loads for you we can since this is a custom report.  You just need to be aware of that when doing week over week comparisons.

**Comment by bevnobev** [comment]
Thanks @michelle-dooley Let me think on that.  If we decide to change to a daily load, how do we go about doing that?

**Comment by michelle-dooley** [comment]
@bevnobev just let us know and we can make the change, it is very simple update to the query to remove the date limit.

**Comment by bevnobev** [comment]
Good Morning @michelle-dooley.  I'd like to change to a daily load.  When we do that, are we able to see daily stats in Domo?  Currently some of the sections don't show daily stats.

**Comment by michelle-dooley** [comment]
Hi @bevnobev - we will make that change for you.  You can change the filter at the top right to "graph by - day" to see daily stats.  You can also click on any graph/card title to drill down into more details about the metrics.

![image](https://user-images.githubusercontent.com/93597659/161592993-4e5531eb-2401-4936-8387-6b9cb42066d7.png)

**Comment by bevnobev** [comment]
@michelle-dooley I drilled down into "Top  reasons emails not retrieved", but didn't see any daily data 

![image](https://user-images.githubusercontent.com/74934207/161611802-e5f6f36a-cecd-4cc8-96c3-c14b8c1211cd.png)

**Comment by mjones-oddball** [comment]
@michelle-dooley Can we set up 30 minutes to review the dashboard with Bev and me so we understand how best to utilize this? We are comparing it to the query in this task description to figure out how the dashboard will replace that.

**Comment by michelle-dooley** [comment]
Yes, I'll get something on our calendars.

**Comment by jerinb-sdi** [comment]
### Ease of use
Chart title | Assets | Pass / Fail
-- | -- | --
Total users who opened emails| [Custom segment](https://analytics.google.com/analytics/web/template?uid=zuTl_gGdTny17cfpnfYBeg) | :green_circle: PASS
Avg. Campaign opens per user| [Custom segment](https://analytics.google.com/analytics/web/template?uid=zuTl_gGdTny17cfpnfYBeg) / [Filtered report](https://analytics.google.com/analytics/web/#/report/content-event-events/a50123418w177519031p176188361/_u.date00=20220220&_u.date01=20220227&explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.eventCategory:email,analytics.eventAction:open/) | :green_circle: PASS
### Findability
Chart title | Assets | Pass / Fail
-- | -- | --
Claims & Appeals - VANotify users |  [Custom segment](https://analytics.google.com/analytics/web/template?uid=hPkSP-KeQI-fEBLxjhMsrw) | :green_circle: PASS
Manage VA Debt - VANotify users |[Custom segment](https://analytics.google.com/analytics/web/template?uid=goyuGCMjTuC_DegE_Luzvw) | :green_circle: PASS
VA Profile - VANotify users | [Custom segment](https://analytics.google.com/analytics/web/template?uid=ZhMHVJCiRJSW_qyqA4PqBQ) | :green_circle: PASS
Avg. session per user for VANotify source | BigQuery metric | :green_circle: PASS
Bounce rate for VANotify sessions | [Custom segment](https://analytics.google.com/analytics/web/template?uid=yV4VXp1iTRSMiwL6d9UQ0Q) | :green_circle: PASS

**Comment by jonathan-epstein13** [comment]
Hello @mjones-oddball and @bevnobev the dashboard has been updated with the all the requested changes and has been QA'd. Let us know what you think. If you have any more questions or concerns, feel free to reach out.

**Comment by michelle-dooley** [comment]
Hi @mjones-oddball and @bevnobev I am going to close this ticket as this work is complete.  If you need anything else please reach out and we will be happy to help!!

**Comment by mjones-oddball** [comment]
Hi @michelle-dooley Thank you so much for the domo updates! It looks great. 2 questions we haven't figured out:
1. How can we share this out? Looks like it's locked down to an access list.
2. We recently launched an email notification and can see that the clicks/referrals are being tracked in [Google Analytics](https://analytics.google.com/analytics/web/#/report/content-event-pages/a50123418w177519031p176188361/explorer-table.plotKeys=%5B%5D&explorer-table.rowCount=10&explorer-table.secSegmentId=analytics.campaign&explorer-table.advFilter=%5B%5B0,%22analytics.pagePath%22,%22RE%22,%22www.va.gov~2Fview-change-dependents~2Fadd-remove-form-21-686c~2Fresume~2F%22,0%5D,%5B0,%22analytics.campaign%22,%22PT%22,%22686c%22,0%5D%5D/). However I'm not seeing it show up under the Landing Page section of our dashboard. The page itself is http://www.va.gov/view-change-dependents/add-remove-form-21-686c/resume/

Will new notifications/campaigns/landing pages automatically show up in the domo dashboard?

CC @bevnobev

**Comment by michelle-dooley** [comment]
Hi @mjones-oddball Glad everything is looking good!!  As far as your questions....
1.  We updated the access level for you, you should be able to share now.  If not let me know.
2. We will automatically get new click events and notification totals for any new campaigns that go out. You can see that the 686c notifications are already filterable on the dashboard level and in the landing page section for the urls we're already tracking (see 1st screenshot).
However, as far as what urls we're looking at for the landing page section, there's no way to automate those updates - you will have to let us know when there's another you'd like added. These urls manually added into the landing page BQ view (2nd screenshot).  If you have another you would like included just let us know and we can get it added.
![image](https://user-images.githubusercontent.com/93597659/167013364-1b0d9f90-24d0-4c2e-a038-a6c6526bd4e6.png)
![image](https://user-images.githubusercontent.com/93597659/167013473-b92ac173-6add-493b-a9d4-77aff74255f0.png)

**Comment by bevnobev** [comment]
Thanks @michelle-dooley.  How do we make that request?  
One other item.  I'm noticing that the campaign landing page numbers for Claims & Appeals seems off between GA and Domo.  [GA ](https://analytics.google.com/analytics/web/?authuser=0#/report/content-event-pages/a50123418w177519031p176188361/_u.date00=20220411&_u.date01=20220504&_.gSectionId=events&explorer-table.plotKeys=%5B%5D&explorer-table.advFilter=%5B%5B0,%22analytics.source%22,%22PT%22,%22vanotify%22,0%5D%5D&explorer-table.secSegmentId=analytics.campaign&_r.drilldown=analytics.pagePath:www.va.gov~2Fclaim-or-appeal-status~2F&explorer-segmentExplorer.segmentId=analytics.source&explorer-graphOptions.primaryConcept=analytics.visits&explorer-graphOptions.compareConcept=analytics.uniqueEventsTrue) shows ~74K and [Domo ](https://va-gov.domo.com/page/2040841289/kpis/details/296331562?pfilters=%5B%7B%22column%22:%22campaign%22,%22operand%22:%22IN%22,%22values%22:%5B%22Form%20526ez%20email%20confirmation%22,%22debt%20notification%22,%22Higher-Level%20Review%20VA%20Form%2020-0996%20Email%20Confirmation%22,%22686c%22%5D,%22dataType%22:%22string%22,%22affectedCardUrns%22:%5B%22907105499%22,%22296331562%22,%221042874468%22,%221031385988%22,%221973179658%22,%22708164314%22%5D,%22label%22:%22campaign%22,%22dataSourceId%22:%221fcd5fbe-add0-4fb2-a387-e302919df2b0%22%7D%5D&refreshPageFilters=true) shows 8k for the same time period.  GA shows events and Domo states users, but that's a huge difference.  Am I missing something?   cc: @mjones-oddball @megansiddle

**Comment by michelle-dooley** [comment]
Hi @bevnobev please complete the update request ticket below.

 https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=michelle-dooley&labels=analytics-insights%2C+analytics-request%2C+kpi-dashboard&template=analytics-request-update-kpi-dashboard.md&title=Analytics+KPI+dashboard+update+request+for+%5BTeam%5D

As far the campaign landing page numbers I believe you are using the wrong GA report.  You are using the Pages report which is showing ALL pageviews while the Domo version is meant to recreate the Landing page report which only counts when the user started their session with that url.  
cc:  @mjones-oddball @megansiddle
